application/src/main/java/uk/nhs/hee/tis/revalidation/messages/RecommendationStatusCheckUpdatedMessageHandler.java

```diff
@@ -92,18 +91,18 @@ private void setRecommendationStatusAndUpdateRepositories(final String recommend
       log.warn("Ignoring an update to an unknown Recommendation: {}", recommendationId);
       return;
     }
+    Recommendation recommendation = optionalRecommendation.get();
+    recommendation.setOutcome(recommendationGmcOutcome);
     if (APPROVED.equals(recommendationGmcOutcome) || REJECTED.equals(recommendationGmcOutcome)) {
-      Recommendation recommendation = optionalRecommendation.get();
-      recommendation.setOutcome(recommendationGmcOutcome);
       recommendation.setRecommendationStatus(COMPLETED);
-      recommendationRepository.save(recommendation);
+    }
+    recommendationRepository.save(recommendation);
 
-      // Find the snapshot from snapshot repository and check the gmcRecommendationId.
-      // If it exists then don't save snapshot else save it
-      if (!doesSnapshotRecommendationExist(recommendation.getGmcNumber(), gmcRecommendationId)) {
-        recommendation.setGmcRevalidationId(gmcRecommendationId);
-        snapshotService.saveRecommendationToSnapshot(recommendation);
-      }
+    // Find the snapshot from snapshot repository and check the gmcRecommendationId.
+    // If it exists then don't save snapshot else save it
+    if (!doesSnapshotRecommendationExist(recommendation.getGmcNumber(), gmcRecommendationId)) {
+      recommendation.setGmcRevalidationId(gmcRecommendationId);
+      snapshotService.saveRecommendationToSnapshot(recommendation);
```

Comment:
Thinking we may only want to save the snapshot when the recommendation is COMPLETED (APPROVED / REJECTED), so it may need to stay inside the `if (APPROVED.equals(recommendationGmcOutcome) || REJECTED.equals(recommendationGmcOutcome)) {}` clause